### PR TITLE
Chore/vrf 538 setup scripts for vrfv

### PR DESCRIPTION
**Pull request by iljapavlovs**
## NOTE:
* Script will delete all existing jobs on the node!
* Currently works only with 0 or 1 VRF Keys on the node! Otherwise, will stop execution!
* Currently possible to fund all nodes with one amount of native tokens
## Commands:
1. If using Docker Compose
   1. create `.env` file in `core/scripts/vrfv2/testnet/docker` (can use `sample.env` file as an example)
   2. go to `core/scripts/vrfv2/testnet/docker` folder and start containers - `docker compose up`
2. Populate `./creds/` folder with relevant credentials for the nodes
3. Ensure that following env variables are set
```
export ETH_URL=
export ETH_CHAIN_ID=
export ACCOUNT_KEY=
```
3. execute from `core/scripts/vrfv2/testnet/setup-envs` folder
```
go run . \
--vrf-primary-node-url=http://localhost:6610 \
--vrf-primary-creds-file ./creds/vrf-primary-node.txt \
--vrf-backup-node-url=http://localhost:6611 \
--vrf-bk-creds-file ./creds/vrf-backup-node.txt \
--bhs-node-url=http://localhost:6612 \
--bhs-creds-file ./creds/bhs-node.txt \
--bhs-backup-node-url=http://localhost:6613 \
--bhs-bk-creds-file ./creds/bhs-backup-node.txt \
--bhf-node-url=http://localhost:6614 \
--bhf-creds-file ./creds/bhf-node.txt \
--num-eth-keys 5 \
--num-vrf-keys 1 \
--sending-key-funding-amount 100000000000000000
```

### CODE REVIEW

File: core/scripts/vrfv2/testnet/setup-envs/main.go [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+		for i := 0; i < toCreate; i++ {
+			output.Reset()
+			newKey := createVRFKey(client, app, output)
+			newKeys = append(newKeys, newKey)
+			fmt.Println("NEW VRF KEYS:", strings.Join(func() (r []string) {
+				for _, k := range newKeys {
+					r = append(r, k.Uncompressed)
+				}
+				return
+			}(), ", "))
```
Comment (makramkd) [inline core/scripts/vrfv2/testnet/setup-envs/main.go:379]: This is being run on every iteration of the loop, maybe only need to run it one afterwards?

File: core/scripts/vrfv2/testnet/setup-envs/main.go [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+	var allVRFKeys []presenters.VRFKeyResource
+	var newKeys []presenters.VRFKeyResource
+
+	vrfKeys := getVRFKeys(client, app, output)
+
+	switch {
+	case len(vrfKeys) == *numVRFKeys:
+		fmt.Println(checkMarkEmoji, "found", len(vrfKeys), "vrf keys on", nodeURL)
+	case len(vrfKeys) > *numVRFKeys:
+		fmt.Println(checkMarkEmoji, "found", len(vrfKeys), "vrf keys on", nodeURL, " which is more than expected")
```
Comment (makramkd) [inline core/scripts/vrfv2/testnet/setup-envs/main.go:364]:
```suggestion
		fmt.Println(xEmoji, "found", len(vrfKeys), "vrf keys on", nodeURL, " which is more than expected")
```

File: core/scripts/vrfv2/testnet/setup-envs/main.go [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+	flagSet.String("output", "exportedvrf.json", "")
+	err := flagSet.Parse([]string{vrfKey.Compressed})
+	helpers.PanicErr(err)
+	err = client.ExportVRFKey(cli.NewContext(app, flagSet, nil))
+	helpers.PanicErr(err)
+	output.Reset()
+}
+
+func importVRFKey(client *clcmd.Shell, app *cli.App, output *bytes.Buffer) {
+	if err := os.WriteFile("vrf-key-password.txt", []byte("twochains"), 0666); err != nil {
```
Comment (makramkd) [inline core/scripts/vrfv2/testnet/setup-envs/main.go:264]: What's this for?

File: core/scripts/vrfv2/testnet/setup-envs/main.go [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+	err := flagSet.Parse([]string{"./job-spec.toml"})
+	helpers.PanicErr(err)
+	err = client.CreateJob(cli.NewContext(app, flagSet, nil))
+	helpers.PanicErr(err)
+	helpers.PanicErr(json.Unmarshal(output.Bytes(), &job))
+	output.Reset()
+}
+
+func exportVRFKey(client *clcmd.Shell, app *cli.App, vrfKey presenters.VRFKeyResource, output *bytes.Buffer) {
+	if err := os.WriteFile("vrf-key-password.txt", []byte("twochains"), 0666); err != nil {
```
Comment (makramkd) [inline core/scripts/vrfv2/testnet/setup-envs/main.go:250]: And this?

File: core/scripts/vrfv2/testnet/docker/db/create-multiple-databases.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
+set -u
+
+function create_user_and_database() {
```
[the remark this answers is not in the document]
Comment (iljapavlovs) [reply core/scripts/vrfv2/testnet/docker/db/create-multiple-databases.sh:6]: this file just creates multiple databases with one instance of Postgres, so you dont need to have separate dbs specified in the docker-compose file

File: core/scripts/vrfv2/testnet/docker/wait-for-others/docker-wait-for-it.sh [hunk 6]
```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
+# From https://github.com/vishnubob/wait-for-it
```
Comment (iljapavlovs) [inline core/scripts/vrfv2/testnet/docker/wait-for-others/docker-wait-for-it.sh:2]: is it ok to use this file in our Chainlink repo? It refers to a utility script which uses MIT license

File: core/scripts/vrfv2/testnet/setup-envs/main.go (hunk 3 above)
Comment (iljapavlovs) [reply core/scripts/vrfv2/testnet/setup-envs/main.go:264]: when importing VRF Key, we need to specify a file where password is stored

File: core/scripts/vrfv2/testnet/setup-envs/main.go (hunk 4 above)
Comment (iljapavlovs) [reply core/scripts/vrfv2/testnet/setup-envs/main.go:250]: when exporting VRF Key, we need to specify a file where password is stored for import action

**Review by makramkd — APPROVED** [review]

**Pull request merged**